packages/block-library/src/block/edit.js

```diff
@@ -284,6 +286,9 @@ export default function ReusableBlockEdit( {
 				onNavigateToEntityRecord:
 					getSettings().onNavigateToEntityRecord,
 				editingMode: _getBlockEditingMode( patternClientId ),
+				hasPatternOverridesSource: !! getBlockBindingsSource(
```

Comment:
Yeah, I was in two minds about it.

My thinking is that this could be maintained to work like this even after the `process.env.IS_GUTENBERG_PLUGIN` code is removed, as a potential `unregisterBindingSource( 'core/pattern-overrides' )` call could result in the same thing.

Actually, I don't think this approach is a good idea, as a WP 6.5 user could re-register pattern overrides, so I think we'll need to add `process.env.IS_GUTENBERG_PLUGIN` everywhere.

Reply:
> The register action is private at the moment, so maybe this solution is ok.

Given that it's still _possible_, shall we just put behind the flag?

Reply:
@getdave It's not possible for third-parties, so it doesn't impact. 

Personally, I don't mind either way.

Reply:
Oh I thought a third party could opt into Private APIs it's just they have to jump through a lot of hoops to do so. 

> Personally, I don't mind either way.

Good enough for me. @talldan can make a call.